heron/instance/src/java/com/twitter/heron/instance/Slave.java

```diff
@@ -179,6 +179,8 @@ public void close() {
     LOG.info("Closing the Slave Thread");
     this.metricsCollector.forceGatherAllMetrics();
     LOG.info("Cleaning up the instance");
-    instance.stop();
+    if (instance != null) {
```

Comment:
Fine with me. I was thinking since it was exposing some corner conditions - it might be worthwhile. Leave it to you.